### DISCUSSION TITLE
Draw a rectangle, the color is red, how not to redraw, you can directly change the color to green

### DISCUSSION THREAD

**Discussion by pcbren1** [pixijs/pixijs#7574] [Q&A] [answered]
Draw a rectangle, the color is red, how not to redraw, you can directly change the color to green
![5555](https://user-images.githubusercontent.com/86771584/124301889-05dd5980-db93-11eb-8a40-3cb0b499560d.png)

**Comment by ivanpopelyshev** [comment, marked as answer]
draw white rectangle. use `g1.tint = 0xff0000`

**Reply by pcbren1** [reply]
When using pixijs to draw PCB, there are many objects in each layer, and the color of each layer is different. There are 100k objects at most. It will take more time to redraw and modify the color. Is there any function to change the color directly

For example, if you change red to green, green can turn yellow

![2021-07-03_084940](https://user-images.githubusercontent.com/86771584/124338793-60e76e80-dbdc-11eb-9f17-3a5347aedc3b.png)

**Reply by ivanpopelyshev** [reply]
Its possible to do, if you know how to make custom shaders for Sprites and Graphics. 

For example, here's shader that inverts all the colors: https://www.pixiplayground.com/#/edit/niqyAgXJD-3hSMYKJPicf

Its not easy as that, because for your case it needs extra uniforms that depend on layer and color swap in glsl.

Alternatively, you can use color-replace filter from pixi-filters on each layer, it will affect performance.

Maybe you actually need paid help for that thing?

**Comment by bigtimebuddy** [comment]
Alternatively, if you just need a box, you can try to use a Sprite with a default texture instead.

```js
const sprite = new PIXI.Sprite(PIXI.Texture.WHITE);
sprite.width = 400;
sprite.height = 50;
sprite.tint = 0xff0000; // change to red
```

**Reply by pcbren1** [reply]
When using pixijs to draw PCB, there are many objects in each layer, and the color of each layer is different. There are 100k objects at most. It will take more time to redraw and modify the color. Is there any function to change the color directly

For example, if you change red to green, green can turn yellow
